core/commands/files/files.go

```diff
@@ -860,8 +961,36 @@ Remove files or directories.
 	},
 }
 
-func getFileHandle(r *mfs.Root, path string, create bool) (*mfs.File, error) {
+func getPrefix(req cmds.Request) (*cid.Prefix, error) {
+	cidVer, cidVerSet, _ := req.Option("cid-version").Int()
+	hashFunStr, hashFunSet, _ := req.Option("hash-fun").String()
```

Comment:
i'm not sure we should use `hash-fun`. ipfs add just uses `--hash`, i'm thinking we should be consistent.

Reply:
@whyrusleeping That won't work.  The hash option is already used in one of the sub-commands and will create a conflict.